hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -2469,17 +2469,17 @@ private void internalInitSubclassPropertyAliasesMap(String path, Iterator proper
 	public Object loadByUniqueKey(
 			String propertyName,
 			Object uniqueKey,
-			SharedSessionContractImplementor session) throws HibernateException {
-		return getAppropriateUniqueKeyLoader( propertyName, session ).loadByUniqueKey( session, uniqueKey );
+			SharedSessionContractImplementor session) {
+		return getAppropriateUniqueKeyLoader( propertyName, session ).load( uniqueKey, session, LockOptions.NONE );
 	}
 
-	private EntityLoader getAppropriateUniqueKeyLoader(String propertyName, SharedSessionContractImplementor session) {
+	protected UniqueEntityLoader getAppropriateUniqueKeyLoader(String propertyName, SharedSessionContractImplementor session) {
```

Comment:
@Sanne I think we might have lost this change:

`getAppropriateUniqueKeyLoader()` needs to be made visible.

Reply:
Ok, I wondered about that. Isn't access to override `createUniqueKeyLoader` good enough for HR ?

Reply:
Ok, I wondered about that. Isn't access to override `createUniqueKeyLoader` good enough for HR ?

Reply:
No i need it.

Reply:
Ok, done. Thanks!